cli/helpers.ts

```diff
@@ -0,0 +1,216 @@
+import * as prettier from "prettier";
+import * as fs from "fs";
+import * as fse from "fs-extra";
+import * as path from "path";
+import chalk from "chalk";
+export const log = console.log;
+import getPort from "get-port";
+
+export enum Package {
+  Frontend = "frontend",
+  Backend = "backend",
+  TopLevel = "toplevel",
+}
+
+export class Ports {
+  frontend: number | undefined = undefined;
+  backend: number | undefined = undefined;
+  db: number | undefined = undefined;
+  setup = async (): Promise<void> => {
+    if (this.backend === undefined) {
+      this.backend = await getPort();
+    }
+    if (this.frontend === undefined) {
+      this.frontend = await getPort();
+    }
+    if (this.db === undefined) {
+      this.db = await getPort();
+    }
+  };
+}
+
+export const FRONTEND_REGEXP = /\NEXT_PUBLIC_GRAPHQL_URL=http:\/\/localhost:\d+\/graphql/;
+export const BACKEND_REGEXP = /\PORT=\d+/g;
+export const DATABASE_URL_REGEXP = /\DATABASE_URL=postgres:\/\/postgres:password@localhost\/postgres/;
+
+export class UnreachableCaseError extends Error {
+  constructor(val: never) {
+    super(`Unreachable case: ${JSON.stringify(val)}`);
+  }
+}
+
+export function copyTemplate(srcDir: string, targetDir: string) {
+  // Copy template, ignoring node_modules
+  try {
+    fse.copySync(srcDir, targetDir, {
+      filter: function (path) {
+        return !(path.indexOf("node_modules") > -1);
+      },
+    });
+    log(chalk.cyan(`Template copied to:\n  ${chalk.cyan.bold(targetDir)}`));
+  } catch {
+    throw new Error(
+      chalk.red(`Error copying template from ${srcDir} to ${targetDir}`)
+    );
+  }
+}
```

Comment:
I would rather use degit to download the code from Github to a temporary directory, and copy over the template directory from there. There are two advantages: we don't have to include special logic to skip untracked files, and we would always get an up-to-date template in case the user hasn't updated the CLI recently.